### PR TITLE
[ET-401] IAC > Unique required meta.

### PR DESCRIPTION
**Pull request by juanfra**
🎫 [ET-401] 
🎥 https://www.loom.com/share/609cdb8577204f079913d6da1a27fc52?from_recorder=1

Modifying some tickets form styles to enable what we need for IAC unique error messages.

### CODE REVIEW

File: src/Tribe/Attendee_Registration/Shortcode.php [hunk 1]
```diff
@@ -1,11 +1,13 @@
...
-* @since 4.10.2
-*/
+ * Provides shortcodes for the attendee registration template.
+ *
+ * @since 4.10.2
+ */
 class Tribe__Tickets__Attendee_Registration__Shortcode {
 	protected $shortcode_name = 'tribe_attendee_registration';
-	protected $params = array();
+	protected $params         = [];
```
Comment (tr1b0t) [inline src/Tribe/Attendee_Registration/Shortcode.php:10]: :no_entry_sign: Missing member variable doc comment

<sub>hash: 697cf1a35d5b52045fbc3f64acabd822</sub>

File: src/Tribe/Attendee_Registration/View.php [hunk 2]
```diff
@@ -3,19 +3,7 @@
...
-	 * @since 4.9
-	 */
-	public function __construct() {
-		$this->set_template_origin( tribe( 'tickets.main' ) );
-		$this->set_template_folder( 'src/views' );
-		$this->set_template_context_extract( true );
-		$this->set_template_folder_lookup( true );
-	}
-
+class Tribe__Tickets__Attendee_Registration__View extends Tribe__Tickets__Editor__Template {
```
Comment (tr1b0t) [inline src/Tribe/Attendee_Registration/View.php:6]: :no_entry_sign: Class name is not valid; consider Tribe_Tickets_Attendee_Registration_View instead

<sub>hash: bdad6d8a6dd00d4263ccc8dd59a428ec</sub>

File: src/Tribe/Attendee_Registration/View.php [hunk 3]
```diff
@@ -37,138 +25,14 @@ public function display_attendee_registration_page() {
...
-			if ( empty( $default_provider ) ) {
-				// One provider per instance.
-				$default_provider[ $q_provider ] = $ticket->provider->class_name;
-			}
-
-			/** @var Tribe__Tickets__Status__Manager $status */
-			$status   = tribe( 'tickets.status' );
-			$provider = $status->get_provider_slug( $ticket->provider->class_name );
+		try {
+			/* @var Tribe__Tickets_Plus__Attendee_Registration__View $view */
```
Comment (tr1b0t) [inline src/Tribe/Attendee_Registration/View.php:29]: :warning: This comment is 50% valid code; is this commented out code?

<sub>hash: 30774923fc13b6466fa167958f4575b3</sub>

File: src/Tribe/Attendee_Registration/View.php [hunk 4]
```diff
@@ -181,41 +45,14 @@ public function display_attendee_registration_shortcode() {
...
-			} elseif ( 'Tribe__Tickets_Plus__Commerce__EDD__Main' === get_class( $post_provider ) ) {
-				/** @var \Tribe__Tickets_Plus__Commerce__EDD__Main $provider */
-				$provider = tribe( 'tickets-plus.commerce.edd' );
-			} else {
-				return;
-			}
-		} catch ( RuntimeException $exception ) {
-			return;
-		}
+			/* @var Tribe__Tickets_Plus__Attendee_Registration__View $view */
```
Comment (tr1b0t) [inline src/Tribe/Attendee_Registration/View.php:49]: :warning: This comment is 50% valid code; is this commented out code?

<sub>hash: be332497d1afc8e63f0192b006f86927</sub>

File: src/Tribe/Attendee_Registration/View.php [hunk 5]
```diff
@@ -230,31 +67,14 @@ public function get_cart_url( $provider ) {
...
-		 *
-		 * @since 4.11.0
-		 *
-		 * @param string $provider A string indicating the desired provider.
-		 *
-		 * @return boolean|object The provider object or boolean false if none found above.
-		 */
-		$provider_obj = apply_filters( 'tribe_attendee_registration_cart_provider', $provider_obj, $provider );
+		try {
+			/* @var Tribe__Tickets_Plus__Attendee_Registration__View $view */
```
Comment (tr1b0t) [inline src/Tribe/Attendee_Registration/View.php:71]: :warning: This comment is 50% valid code; is this commented out code?

<sub>hash: 0d7fb42ab55e622addafbc29eb54c23c</sub>

File: src/Tribe/Attendee_Registration/View.php [hunk 6]
```diff
@@ -268,47 +88,13 @@ public function get_cart_provider( $provider ) {
...
-		/**
-		 * Allow providers to include their own strings/suffixes.
-		 *
-		 * @since 4.10.4
-		 *
-		 * @param array $provider_classes In the format of: $provider -> class suffix.
-		 */
-		$provider_classes = apply_filters( 'tribe_attendee_registration_form_classes', [] );
+		try {
+			/* @var Tribe__Tickets_Plus__Attendee_Registration__View $view */
```
Comment (tr1b0t) [inline src/Tribe/Attendee_Registration/View.php:92]: :warning: This comment is 50% valid code; is this commented out code?

<sub>hash: c3e7f089f4644e0960af9036c5bf6f2b</sub>

File: src/Tribe/Commerce/PayPal/Main.php [hunk 7]
```diff
@@ -3021,10 +3083,12 @@ public function get_attendee( $attendee, $post_id = 0 ) {
 	 */
 	protected function get_cancelled( $ticket_id ) {
 		$denied_orders = Tribe__Tickets__Commerce__PayPal__Order::find_by( array(
-			'ticket_id'   => $ticket_id,
-			'post_status' => Tribe__Tickets__Commerce__PayPal__Stati::$denied,
+			'ticket_id'      => $ticket_id,
+			'post_status'    => Tribe__Tickets__Commerce__PayPal__Stati::$denied,
 			'posts_per_page' => -1,
-		) );
+		), [
```
Comment (tr1b0t) [inline src/Tribe/Commerce/PayPal/Main.php:3089]: * :no_entry_sign: Multi-line function call not indented correctly; expected 12 spaces but found 8
* :no_entry_sign: Only one argument is allowed per line in a multi-line function call

<sub>hash: 888f415104e4ee62a3e4794826bc12bd</sub>

File: src/Tribe/Commerce/PayPal/Main.php [hunk 8]
```diff
@@ -3021,10 +3083,12 @@ public function get_attendee( $attendee, $post_id = 0 ) {
...
+			'items',
+		] );
```
Comment (tr1b0t) [inline src/Tribe/Commerce/PayPal/Main.php:3091]: :no_entry_sign: Closing parenthesis of a multi-line function call must be on a line by itself

<sub>hash: b55121a51e72f0c393d842ae0d3d93c6</sub>

File: src/Tribe/Commerce/PayPal/Order.php [hunk 9]
```diff
@@ -303,25 +312,26 @@ public function hydrate_from_transaction_data( array $transaction_data ) {
...
 	 * @type int    $ticket_id ID, or array of IDs, of the ticket(s) Orders should be related to.
 	 * }
+	 * @param null|array $fields List of fields to hydrate, or null for all.
 	 *
 	 * @return Tribe__Tickets__Commerce__PayPal__Order[]
 	 */
-	public static function find_by( array $args = array() ) {
-		$args = wp_parse_args( $args, array(
+	public static function find_by( array $args = [], $fields = null ) {
+		$args = wp_parse_args( $args, [
```
Comment (tr1b0t) [inline src/Tribe/Commerce/PayPal/Order.php:327]: * :no_entry_sign: Opening parenthesis of a multi-line function call must be the last content on the line
* :no_entry_sign: Only one argument is allowed per line in a multi-line function call

<sub>hash: c65921623a06d13c6a9036354ae28e3a</sub>

File: src/Tribe/Commerce/PayPal/Order.php [hunk 10]
```diff
@@ -303,25 +312,26 @@ public function hydrate_from_transaction_data( array $transaction_data ) {
...
 			'post_type'   => Tribe__Tickets__Commerce__PayPal__Main::ORDER_OBJECT,
 			'post_status' => 'any',
 			'meta_key'    => self::$meta_prefix . 'payment_date',
 			'meta_type'   => 'DATETIME',
 			'order'       => 'DESC',
 			'orderby'     => 'meta_value',
-		) );
+		] );
```
Comment (tr1b0t) [inline src/Tribe/Commerce/PayPal/Order.php:334]: :no_entry_sign: Closing parenthesis of a multi-line function call must be on a line by itself

<sub>hash: f77b3641a072e5af5cc28c33ebfb24ea</sub>

File: src/Tribe/Editor/Blocks/Rsvp.php [hunk 11]
```diff
@@ -1,6 +1,6 @@
 <?php
-class Tribe__Tickets__Editor__Blocks__Rsvp
-extends Tribe__Editor__Blocks__Abstract {
+
+class Tribe__Tickets__Editor__Blocks__Rsvp extends Tribe__Editor__Blocks__Abstract {
```
Comment (tr1b0t) [inline src/Tribe/Editor/Blocks/Rsvp.php:3]: * :no_entry_sign: Missing doc comment for class Tribe__Tickets__Editor__Blocks__Rsvp
* :no_entry_sign: Class name is not valid; consider Tribe_Tickets_Editor_Blocks_Rsvp instead

<sub>hash: 1ce86266d216e8e844be00dd36a61f3b</sub>

File: src/Tribe/Editor/REST/V1/Endpoints/Single_Ticket.php [hunk 12]
```diff
@@ -263,6 +263,7 @@ public function add_ticket( WP_REST_Request $request, $nonce_action ) {
 			'ticket_end_date' => $body['end_date'],
 			'ticket_end_time' => $body['end_time'],
 			'ticket_sku' => $body['sku'],
+			'ticket_iac' => $body['iac'],
```
Comment (tr1b0t) [inline src/Tribe/Editor/REST/V1/Endpoints/Single_Ticket.php:266]: :warning: Array double arrow not aligned correctly; expected 14 space(s) between "'ticket_iac'" and double arrow, but found 1.

<sub>hash: aa596115452417c577fb4d42482a08d9</sub>

File: src/Tribe/REST/V1/Post_Repository.php [hunk 13]
```diff
@@ -962,6 +962,11 @@ protected function build_attendee_data( array $attendee, $context = 'default' )
 					'email'           => Tribe__Utils__Array::get( $attendee, 'holder_email', Tribe__Utils__Array::get( $attendee, 'purchaser_email', '' ) ),
 					'checked_in'      => $checked_in,
 					'checkin_details' => $checkin_details,
+
+					// Show Attendee flags.
+					// @todo Make these live in future IAC work.
+					'is_subscribed' => false,
```
Comment (tr1b0t) [inline src/Tribe/REST/V1/Post_Repository.php:968]: :warning: Array double arrow not aligned correctly; expected 3 space(s) between "'is_subscribed'" and double arrow, but found 1.

<sub>hash: fdc92cb4020865001f28fb8f9d3dd927</sub>

File: src/Tribe/REST/V1/Post_Repository.php [hunk 14]
```diff
@@ -962,6 +962,11 @@ protected function build_attendee_data( array $attendee, $context = 'default' )
...
+					'is_purchaser'  => true,
```
Comment (tr1b0t) [inline src/Tribe/REST/V1/Post_Repository.php:969]: :warning: Array double arrow not aligned correctly; expected 4 space(s) between "'is_purchaser'" and double arrow, but found 2.

<sub>hash: 769f38b9479112a21bafcf40f788c767</sub>

File: src/Tribe/RSVP.php [hunk 15]
```diff
@@ -2037,44 +2074,50 @@ public function get_attendee( $attendee, $post_id = 0 ) {
...
+				'attendee_id'        => $attendee->ID,
+				'security'           => $security,
+				'product_id'         => $product_id,
+				'check_in'           => $checkin,
+				'order_status'       => $status,
+				'order_status_label' => $status_label,
+				'user_id'            => $user_id,
+				'ticket_sent'        => $ticket_sent,
+
+			// Fields for Email Tickets.
```
Comment (tr1b0t) [inline src/Tribe/RSVP.php:2091]: :no_entry_sign: Array item not aligned correctly; expected 16 spaces but found 12

<sub>hash: 904a5ae7bef2402282b7843d33673125</sub>

File: src/Tribe/RSVP.php [hunk 16]
```diff
@@ -2037,44 +2074,50 @@ public function get_attendee( $attendee, $post_id = 0 ) {
...
+				'event_id'           => get_post_meta( $attendee->ID, self::ATTENDEE_EVENT_KEY, true ),
```
Comment (tr1b0t) [inline src/Tribe/RSVP.php:2092]: :no_entry_sign: Multi-line array item not aligned correctly; expected 20 spaces, but found 16

<sub>hash: e0192c5aef3a0b546f84616534fa6c35</sub>

**Review by tr1b0t — COMMENTED** [review]
:no_entry_sign: 12 error(s)
:warning: 7 warning(s)

**Review by sc0ttkclark — APPROVED** [review]

**Pull request merged**